### ISSUE TITLE
App created via the UI doesn't respect App in any namespace settings

### ISSUE THREAD

**Issue by OpenGuidou** [argoproj/argo-cd#20820]
Checklist:

* [ ] I've searched in the docs and FAQ for my answer: https://bit.ly/argocd-faq.
* [ ] I've included steps to reproduce the bug.
* [ ] I've pasted the output of `argocd version`.

**Describe the bug**

When creating an app from the UI, it is not possible to provide the namespace in which the app will be created. It's always created in argo-cd's namespace.

Even when going to the yaml view and setting the namespace, it's overrided and created in the argo-cd namespace anyway.

Furthermore, it is not checking the project settings to know if it's ok to deploy in Argo-cd namespace. I've set the `spec.sourceNamespaces` in the project to reference only the other namespace (not the argo-cd one), and the app is created without issue.

**To Reproduce**

1. Create a namespace to store the argocd apps: `not-argo-namespace`
2. Deploy Argo-CD with the app and appset in any namespace settings, with `application-namespaces=not-argo-namespace` preventing to use argo-cd namespace
4. Set the `spec.sourceNamespace` in the default project to `not-argo-namespace`
5. Create an app from the UI

**Expected behavior**

We should be able to select the authorized namespaces in which to create the App.
If by any mean we set one not in the application-namespaces nor in the project sourceNamespace, it should be rejected.

**Version**

2.13

**Logs**

**Comment by Mangaal** [comment]
The behavior you are observing is actually due to the naming convention required when creating applications in a specific namespace via the UI. As per the documentation, the application name must follow the namespace/app format to indicate the target namespace explicitly.
Please refer to this doc[ app-any-namespace](https://argo-cd.readthedocs.io/en/stable/operator-manual/app-any-namespace/#using-the-cli) .

**Comment by OpenGuidou** [comment]
Thanks for the info.
It's not optimal in terms of user experience, but that works.

However, there should still be the check on the allowed namespaces in which the user can deploy the application.

**Comment by Mangaal** [comment]
I agree, the user experience could be better. Since this behavior is working as intended, I suggest changing the label to feature instead of a bug. Adding a check for allowed namespaces would be a great enhancement!

**Comment by OpenGuidou** [comment]
I would rather fix the missing check as a bug, and provide a better user experience via an additional option to provide the application namespace in the UI and the CLI

**Comment by Mangaal** [comment]
for cli, you can use the flag  --app-namespace

**Comment by OpenGuidou** [comment]
Summary of contributor meeting on the 21st of November 2024:

* UI enhancement is going to be handled in #20836 
* To restrict the access to argocd namespace, it's possible to do it via RBAC:
```yaml
p, role:myspecificrole, applications, *, */argocd/*, deny
p, role:myspecificrole, applicationsets, *, */argocd/*, deny
```
I confirm it's working
* Project restriction is indeed not enforced for argocd namespace. It's adressed in #20827 and we'd like more historical information before progressing further. @jannfis: Could you share why the argocd namespace was always bypassed in project restriction before ? Would it be valid to enforce it now ? cc @crenshaw-dev 
* Cluster secret namespace setting: *Not* listing argocd secret in the "in-cluster" secret does not prevent from deploying resources there. This should be addressed as well
